src/runtime/runtime.cs

```diff
@@ -166,27 +127,97 @@ internal static Version PyVersion
         /// <summary>
         /// Initialize the runtime...
         /// </summary>
-        internal static void Initialize(bool initSigs = false)
+        /// <remarks>Always call this method from the Main thread.  After the 
+        /// first call to this method, the main thread has acquired the GIL.</remarks>
+        internal static void Initialize(bool initSigs = false, ShutdownMode mode = ShutdownMode.Default, bool fromPython = false)
         {
+            if (_isInitialized)
+            {
+                return;
+            }
+            _isInitialized = true;
```

Comment:
Probably not an issue in this PR, but static methods generally should be thread-safe.

Reply:
Addressing this is not necessary for acceptance.

Reply:
I think it's to avoid double initialization of the runtime. With domain reloads we can't rely anymore on `Py_IsInitialized`.